app/controllers/posts_controller.rb

```diff
@@ -1,19 +1,33 @@
 class PostsController < ApplicationController
   before_action :set_new_post, only: %i[create]
-  before_action :set_post, only: %i[show]
+  before_action :set_post, only: %i[show edit update accept reject destroy]
+
+  def index
+    # get all posts if current user is admin
+    # get only user's posts if current user is not admin
```

Comment:
Comment is very good practice but sometimes let the code speak for itself, you can read some best practices to write comment:

https://stackoverflow.blog/2021/12/23/best-practices-for-writing-code-comments/